src/test/java/org/springframework/samples/petclinic/cache/HybridCacheIntegrationTest.java

```diff
@@ -0,0 +1,503 @@
+        reset(vetRepository, ownerRepository, petRepository, petTypeRepository, specialtyRepository, visitRepository);
+        setupAllMockData();
+    }
+
+    /**
+     * Test data provider for parameterized cache tests (collection endpoints)
+     */
+    static Stream<Arguments> cacheTestData() {
+        return Stream.of(
+            Arguments.of("vets", "/api/vets", "vets", "\"id\":1", "\"id\":2"),
+            Arguments.of("owners", "/api/owners", "owners", "\"id\":1", "\"id\":2"),
+            Arguments.of("pets", "/api/pets", "pets", "\"id\":1", "\"id\":2"),
+            Arguments.of("petTypes", "/api/pettypes", "petTypes", "\"id\":1", "\"id\":2"),
+            Arguments.of("specialties", "/api/specialties", "specialties", "\"id\":1", "\"id\":2"),
+            Arguments.of("visits", "/api/visits", "visits", "\"id\":1", "\"id\":2")
+        );
+    }
+
+    /**
+     * Test data provider for parameterized cache tests (single entity endpoints)
+     */
+    static Stream<Arguments> singleEntityCacheTestData() {
+        return Stream.of(
+            Arguments.of("vets", "/api/vets/1", "vetById", "\"id\":1", "Test"),
+            Arguments.of("owners", "/api/owners/1", "ownerById", "\"id\":1", "John"),
+            Arguments.of("pets", "/api/pets/1", "petById", "\"id\":1", "Fluffy"),
+            Arguments.of("petTypes", "/api/pettypes/1", "petTypeById", "\"id\":1", "Cat"),
+            Arguments.of("specialties", "/api/specialties/1", "specialtyById", "\"id\":1", "Dentistry"),
+            Arguments.of("visits", "/api/visits/1", "visitById", "\"id\":1", "Regular checkup")
+        );
+    }
+
+    @ParameterizedTest(name = "testHybridCacheCoordination_{0}")
+    @MethodSource("cacheTestData")
+    void testHybridCacheCoordination(String entityName, String endpoint, String cacheName, String expectedId1, String expectedId2) {
+        // 1. Populate both cache layers
+        ResponseEntity<String> response1 = restTemplate.getForEntity(endpoint, String.class);
+        assertThat(response1.getStatusCode().is2xxSuccessful()).isTrue();
+
+        // 2. Verify both cache layers are populated
+        assertThat(findL1KeysForCache(cacheName)).isNotEmpty();
+        assertThat(findL2KeysForCache(cacheName)).isNotEmpty();
+
+        // 3. Verify caching is working (should be fast L1 hit)
+        clearAllRepositoryInvocations();
+        ResponseEntity<String> response2 = restTemplate.getForEntity(endpoint, String.class);
+        assertThat(response2.getStatusCode().is2xxSuccessful()).isTrue();
+        assertThat(response2.getBody()).contains(expectedId1).contains(expectedId2);
+        verifyNoRepositoryInteractions();
+
+        // 4. Clear L1 cache
+        clearL1Cache(cacheName);
+
+        // 5. Check L2 is working
+        clearAllRepositoryInvocations();
+        ResponseEntity<String> response3 = restTemplate.getForEntity(endpoint, String.class);
+        assertThat(response3.getStatusCode().is2xxSuccessful()).isTrue();
```

Comment:
Probably it's clear if we check body here and in subseqent calls and tests as well, not only status.